src/Microsoft.AspNetCore.Mvc.RazorPages/PageContextAttribute.cs

```diff
@@ -0,0 +1,16 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+
+namespace Microsoft.AspNetCore.Mvc.RazorPages
+{
+    /// <summary>
+    /// Specifies that a Razor Page model property should be set with the current <see cref="PageContext"/> when creating
+    /// the model instance. The property must have a public set method.
+    /// </summary>
+    [AttributeUsage(AttributeTargets.Property, AllowMultiple = false, Inherited = true)]
+    public class PageContextAttribute : Attribute
```

Comment:
Should this go in `.Infrastructure`? Do whatever we did with `ActionContextAttribute`.